packages/frontend/src/components/claims/chain-overview/index.tsx

```diff
@@ -0,0 +1,119 @@
+import classNames from "@/src/utils/classes";
+import type { ChainWithClaimsData } from "..";
+import { Typography } from "@/src/ui/typography";
+import { CHAIN_DATA, SUPPORTED_CHAIN_ICONS } from "@/src/commons";
+import { type SupportedChain } from "@metrom-xyz/sdk";
+import { Button } from "@/src/ui/button";
+import { useTranslations } from "next-intl";
+import {
+    useAccount,
+    usePublicClient,
+    useSimulateContract,
+    useWriteContract,
+} from "wagmi";
+import { metromAbi } from "@metrom-xyz/contracts/abi";
+import { parseUnits } from "viem";
+import { useCallback, useState } from "react";
+
+import styles from "./styles.module.css";
+
+interface ChainOverviewProps {
+    className?: string;
+    chainWithClaimsData: ChainWithClaimsData;
+}
+
+export function ChainOverview({
+    className,
+    chainWithClaimsData,
+}: ChainOverviewProps) {
+    const t = useTranslations("claims");
+    const { address: account } = useAccount();
+    const publicClient = usePublicClient();
+    const { writeContractAsync } = useWriteContract();
+
+    const [claiming, setClaiming] = useState(false);
+    const [claimed, setClaimed] = useState(false);
+
+    const ChainIcon =
+        SUPPORTED_CHAIN_ICONS[chainWithClaimsData.chain.id as SupportedChain];
+
+    const {
+        data: simulatedClaimAll,
+        isLoading: simulatingClaimAll,
+        isError: simulateClaimAllError,
+        error,
+    } = useSimulateContract({
+        abi: metromAbi,
+        address:
+            CHAIN_DATA[chainWithClaimsData.chain.id as SupportedChain].contract
+                .address,
+        functionName: "claimRewards",
+        args: [
+            !account
+                ? []
+                : chainWithClaimsData.claims.map((claim) => {
+                      console.log(claim.amount);
```

Comment:
Console log here.

Reply:
Addressed in fc2090d0f245c01946f822083b8adbd743cd8f78